semantic_conventions/trace/messaging.yaml

```
@@ -110,27 +173,66 @@ groups:
     span_kind: consumer
     brief: 'Semantic convention for a consumer of messages received from a messaging system'
     attributes:
-      - id: operation
-        type:
-          allow_custom_values: false
-          members:
-            - id: receive
-              value: "receive"
-            - id: process
-              value: "process"
-        brief: >
-          A string identifying the kind of message consumption as defined in the
-          [Operation names](#operation-names) section above.
-          If the operation is "send", this attribute MUST NOT be set, since the
-          operation can be inferred from the span kind in that case.
-      - id: consumer_id
+      - id: consumer.id
         type: string
         brief: >
           The identifier for the consumer receiving a message. For Kafka, set it to
-          `{messaging.kafka.consumer_group} - {messaging.kafka.client_id}`, if both are present, or only
-          `messaging.kafka.consumer_group`. For brokers, such as RabbitMQ and Artemis, set it to the `client_id`
+          `{messaging.kafka.consumer.group} - {messaging.kafka.client_id}`, if both are present, or only
+          `messaging.kafka.consumer.group`. For brokers, such as RabbitMQ and Artemis, set it to the `client_id`
           of the client consuming the message.
         examples: 'mygroup - client-6'
+      - ref: messaging.source.name
+        requirement_level:
+          conditionally_required: If the value applies to all messages in the batch.
+      - id: source.kind
+        type:
+          allow_custom_values: true
+          members:
+            - id: queue
+              value: "queue"
+              brief: "A message received from a queue"
+            - id: topic
+              value: "topic"
+              brief: "A message received from a topic"
+        requirement_level:
+          conditionally_required: If the message source is either a `queue` or `topic`.
+        brief: 'The kind of message source'
+      - id: source.template
+        type: string
+        brief: 'Low cardinality field representing messaging source'
+        requirement_level:
+          conditionally_required: >
+            If available. Instrumentations MUST NOT use `messaging.source.name` as template
+            unless low-cardinality of source name is guaranteed.
+        examples: ['/customers/{customerId}']
+        note: >
+          Source names could be constructed from templates.
+          An example would be a source name involving a user name or product id.
+          Although the source name in this case is of high cardinality,
+          the underlying template is of low cardinality and can be effectively
+          used for grouping and aggregation.
+      - id: source.temporary
+        type: boolean
+        requirement_level:
+          recommended: when supported by messaging system and only if the source is temporary. If missing, assumed to be false.
```

```suggestion
          recommended: When supported by messaging system and only if the source is temporary. When missing, the value is assumed to be `false`.
```
So it's consistent with `destination.temporary` above :)